### PR TITLE
[ENH] DB fragment long-running jobs cleanup

### PR DESCRIPTION
**Pull request by rksh**
- Building out a centralized ExecutorService
- Creating a one-at-a-time by job type threadpool for these long-running jobs (and a queue of waiting jobs)
- keeping uploads from happening while these are running
- Swapping out (deprecated) ProgressDialog for the progress panel we use for uploads
- Removing leaky definitions and replacing them with runnables for Backup-DB, Export GPX, and Export M8b tasks.
- updating layour for transfer panel
- Standardizing these pogress-publishing, long-running tasks

TODO: translate new strings!
Getting this out there for review because it's gotten so large.

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java [hunk 1]
```diff
@@ -480,13 +436,13 @@ public void handleDialog(final int dialogId) {
             case CSV_RUN_DIALOG: {
                 // actually need this Activity context, for dialogs
                 ObservationUploader observationUploader = new ObservationUploader(getActivity(),
-                        ListFragment.lameStatic.dbHelper, DataFragment.this, true, false, true);
+                        ListFragment.lameStatic.dbHelper, null, true, false, true);
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java:439]: double checking: is the observationuploader's super ok with a null listener?

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/GpxExportRunnable.java [hunk 2]
```diff
@@ -139,81 +132,75 @@ protected long writeSegmentsWithCursor(final FileWriter writer, final Cursor cur
...
         return lineCount;
     }
 
+    @Override
+    protected void onPreExecute() {
+        activity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                //ALIBI: Android like killing long-running tasks like this if you let the screen shut off
+                activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/GpxExportRunnable.java:157]: ah, clever

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/GpxExportRunnable.java [hunk 3]
```diff
@@ -139,81 +132,75 @@ protected long writeSegmentsWithCursor(final FileWriter writer, final Cursor cur
...
-                pd.dismiss();
-            }
+            activity.runOnUiThread(new Runnable() {
+                              @Override
+                              public void run() {
+                                  clearProgressDialog();
+                              }
+                          });
             // fire share intent?
+            activity.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/GpxExportRunnable.java:177]: If there's an error will the code path get here? or will result be null and the screen on flag won't clear (or other some such case)?

**Review by bobzilladev — APPROVED** [review]
A few questions, but nothing dire. Thanks for breaking this up into smaller chunks!

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/GpxManagementActivity.java [hunk 4]
```diff
@@ -185,8 +190,9 @@ public void handleDialog(int dialogId) {
     private boolean exportRouteGpxFile(long runId) {
         final long totalRoutePoints = ListFragment.lameStatic.dbHelper.getRoutePointCount(runId);
         if (totalRoutePoints > 1) {
-            new AsyncGpxExportTask(this, this,
-                    pd, runId).execute(totalRoutePoints);
+            ExecutorService es = Executors.newFixedThreadPool(1);
```
[the remark this answers is not in the document]
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/GpxManagementActivity.java]: great catch - missed that in the recator!

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/GpxExportRunnable.java (hunk 2 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/GpxExportRunnable.java:153]: it's not foolproof, but helps on compliant devices. (not SMSN)

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/GpxExportRunnable.java (hunk 3 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/GpxExportRunnable.java]: arg, smart editor got too smart for me. fixed.

**Comment by rksh** [comment]
added language files and addressed feedback.

**Pull request merged**